### PR TITLE
Add support for trustedScoringSignalsUrl

### PR DESCRIPTION
**Pull request by taymonbeal**
Fixes: #23

### CODE REVIEW

File: frame/auction.ts [hunk 1]
```diff
@@ -39,3 +62,48 @@ function randomToken() {
...
+  hostname: string,
+  renderingUrls: ReadonlySet<string>
+) {
+  const url = new URL(baseUrl);
+  url.searchParams.append("hostname", hostname);
+  url.searchParams.append(
+    "keys",
+    [...renderingUrls].map(encodeURIComponent).join(",")
+  );
+  const response = await tryFetchJson(url.href);
```
Comment (jeffkaufman) [inline frame/auction.ts:77]: This will not include credentials because it is cross origin, right?

(not including them is the behavior we want)

File: frame/auction.ts [hunk 2]
```diff
@@ -39,3 +62,48 @@ function randomToken() {
...
+  switch (response.status) {
+    case FetchJsonStatus.OK: {
+      const signals = response.value;
+      if (!isKeyValueObject(signals)) {
+        logError(`Expected JSON object but received ${jsonTypeOf(signals)}`);
```
Comment (jeffkaufman) [inline frame/auction.ts:82]: Would it be safe to log the whole thing?

File: frame/auction.ts (hunk 1 above)
Comment (jeffkaufman) [inline frame/auction.ts:77]: We should be verifying that the response has `X-ALLOW: fledge`

**Review by jeffkaufman — APPROVED** [review]

File: frame/auction.ts (hunk 2 above)
Comment (taymonbeal) [reply frame/auction.ts:82]: Yes; it can't leak through to the parent origin. I figured that it's better to keep log messages short and if developers want to know the actual value they can look in the network tab, but I could be convinced otherwise.

File: frame/auction.ts (hunk 1 above)
Comment (taymonbeal) [reply frame/auction.ts:77]: Thanks for pointing this out. It will generally be cross-origin, although it could be same-origin if someone self-hosts the frame. I've now added tests that make this explicit, and in #62 am making it explicit in the implementation and closing the same-origin edge case.

Comment (taymonbeal) [reply frame/auction.ts:77]: Doing this in #62.
